src/components/menus/SheetFilter.vue

```diff
@@ -119,55 +127,69 @@ export default {
       tags: [],
       currentTag: null,
       showAllTags: false,
+      filteredSongs: [],
     };
   },
   async mounted() {
-    this.sort();
     this.tags = await getTags();
+    this.sort();
   },
   watch: {
     searchTerms() {
-      if (!this.searchTerms) {
-        this.$emit("sorted", this.filteredSongs);
-        return;
-      }
-      const term = this.searchTerms?.toLowerCase();
-      const isMatch = (s) => {
-        return (
-          s.title?.toLowerCase().includes(term) ||
-          s.subtitle?.toLowerCase().includes(term) ||
-          s.artist?.toLowerCase().includes(term)
-        );
-      };
-      this.$emit("sorted", this.filteredSongs.filter(isMatch));
+      this.playEffect("ui/click");
+      this.sort();
     },
     showSearch() {
+      this.playEffect("ui/click");
       if (this.showSearch) {
         this.showSort = false;
       } else {
         this.searchTerms = null;
       }
     },
     showSort() {
+      this.playEffect("ui/click");
       if (this.showSort) {
         this.showSearch = false;
       }
     },
+    showAllTags() {
+      this.playEffect("ui/click");
+    },
     currentTag() {
-      this.$emit("sorted", this.filteredSongs);
+      this.playEffect("ui/slide2");
+      this.sort();
+    },
+    songs() {
+      this.sort();
     },
   },
   computed: {
-    filteredSongs() {
-      return this.currentTag
-        ? this.songs.filter((e) => e.tags.includes(this.currentTag))
-        : this.songs;
-    },
     sortIcon() {
       return this.reverseSort ? "arrow-up" : "arrow-down";
     },
   },
   methods: {
+    emitResult() {
+      if (this.currentTag)
+        this.filteredSongs = this.filteredSongs.filter((e) =>
+          e.tags.includes(this.currentTag)
+        );
```

Comment:
Nit: you might want to add a lint to force braces around `if`s and `for`s and `while`s and `do while`s etc. because it's easy to accidentally try to add a statement before/after the single statement.

Reply:
I see, that makes sense, I'll see how can I do this...

Reply:
I've found the rules [here](https://eslint.org/docs/2.0.0/rules/curly), but I might need a bit of help on putting it inside `eslintrc.json` 🥺